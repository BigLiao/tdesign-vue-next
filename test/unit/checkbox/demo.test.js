import { mount } from '@vue/test-utils';

// unit test for component in examples.
describe('Checkbox', () => {
  it('base demo works fine', () => {
    const demo = require('@/examples/checkbox/demos/base.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
});
