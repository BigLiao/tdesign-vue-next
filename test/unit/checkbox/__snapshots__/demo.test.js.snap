// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox base demo works fine 1`] = `
<div>
  <label
    class="t-checkbox"
  >
    <input
      class="t-checkbox__former"
      name=""
      type="checkbox"
      value=""
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      未选中项
    </span>
  </label>
   
  <label
    class="t-checkbox"
  >
    <input
      class="t-checkbox__former"
      name=""
      type="checkbox"
      value=""
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      未选悬停项
    </span>
  </label>
   
  <label
    class="t-checkbox t-is-checked"
  >
    <input
      class="t-checkbox__former"
      name=""
      type="checkbox"
      value=""
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      选中项
    </span>
  </label>
   
  <label
    class="t-checkbox t-is-disabled"
  >
    <input
      class="t-checkbox__former"
      disabled=""
      name=""
      type="checkbox"
      value=""
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      未选禁用项
    </span>
  </label>
   
  <label
    class="t-checkbox t-is-checked t-is-disabled"
  >
    <input
      class="t-checkbox__former"
      disabled=""
      name=""
      type="checkbox"
      value=""
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      选中禁用项
    </span>
  </label>
</div>
`;

exports[`Checkbox checkbox group demo works fine 1`] = `
<div
  class="tdesign-checkbox-group"
>
  <div>
    <div
      class="t-checkbox-group"
    >
      <label
        class="t-checkbox"
      >
        <input
          class="t-checkbox__former"
          name="city"
          type="checkbox"
          value="bj"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          北京
        </span>
      </label>
      <label
        class="t-checkbox"
      >
        <input
          class="t-checkbox__former"
          name="city"
          type="checkbox"
          value="sh"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          上海
        </span>
      </label>
      <label
        class="t-checkbox"
      >
        <input
          class="t-checkbox__former"
          name="city"
          type="checkbox"
          value="gz"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          广州
        </span>
      </label>
      <label
        class="t-checkbox"
      >
        <input
          class="t-checkbox__former"
          name="city"
          type="checkbox"
          value="sz"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          深圳
        </span>
      </label>
    </div>
  </div>
   
  <div>
    <div
      class="t-checkbox-group"
    >
      <label
        class="t-checkbox t-is-checked"
      >
        <input
          class="t-checkbox__former"
          name=""
          type="checkbox"
          value="bj"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          北京
        </span>
      </label>
      <label
        class="t-checkbox"
      >
        <input
          class="t-checkbox__former"
          name=""
          type="checkbox"
          value="sh"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          上海
        </span>
      </label>
      <label
        class="t-checkbox"
      >
        <input
          class="t-checkbox__former"
          name=""
          type="checkbox"
          value="gz"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          广州
        </span>
      </label>
      <label
        class="t-checkbox t-is-checked"
      >
        <input
          class="t-checkbox__former"
          name=""
          type="checkbox"
          value="sz"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          深圳
        </span>
      </label>
    </div>
  </div>
   
  <div>
    <div
      class="t-checkbox-group"
    >
      <label
        class="t-checkbox t-is-disabled"
      >
        <input
          class="t-checkbox__former"
          disabled=""
          name=""
          type="checkbox"
          value="bj"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          北京
        </span>
      </label>
      <label
        class="t-checkbox t-is-disabled"
      >
        <input
          class="t-checkbox__former"
          disabled=""
          name=""
          type="checkbox"
          value="sh"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          上海
        </span>
      </label>
      <label
        class="t-checkbox t-is-checked t-is-disabled"
      >
        <input
          class="t-checkbox__former"
          disabled=""
          name=""
          type="checkbox"
          value="gz"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          广州
        </span>
      </label>
      <label
        class="t-checkbox"
      >
        <input
          class="t-checkbox__former"
          name=""
          type="checkbox"
          value="sz"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          深圳
        </span>
      </label>
    </div>
  </div>
   
  <div>
    <div
      class="t-checkbox-group"
    >
      <label
        class="t-checkbox"
      >
        <input
          class="t-checkbox__former"
          name=""
          type="checkbox"
          value="bj"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          北京
        </span>
      </label>
       
      <label
        class="t-checkbox"
      >
        <input
          class="t-checkbox__former"
          name=""
          type="checkbox"
          value="sh"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          上海
        </span>
      </label>
       
      <label
        class="t-checkbox"
      >
        <input
          class="t-checkbox__former"
          name=""
          type="checkbox"
          value="gz"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          广州
        </span>
      </label>
       
      <label
        class="t-checkbox t-is-checked t-is-disabled"
      >
        <input
          class="t-checkbox__former"
          disabled=""
          name=""
          type="checkbox"
          value="sz"
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          深圳
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`Checkbox indeterminate demo works fine 1`] = `
<div
  class="tdesign-checkbox-indeterminate"
>
  <label
    class="t-checkbox t-is-indeterminate"
  >
    <input
      class="t-checkbox__former"
      name=""
      type="checkbox"
      value=""
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      全选
    </span>
  </label>
   
  <br />
   
  <div
    class="t-checkbox-group"
  >
    <label
      class="t-checkbox t-is-checked"
    >
      <input
        class="t-checkbox__former"
        name="city"
        type="checkbox"
        value="bj"
      />
      <span
        class="t-checkbox__input"
      />
      <span
        class="t-checkbox__label"
      >
        北京
      </span>
    </label>
    <label
      class="t-checkbox"
    >
      <input
        class="t-checkbox__former"
        name="city"
        type="checkbox"
        value="sh"
      />
      <span
        class="t-checkbox__input"
      />
      <span
        class="t-checkbox__label"
      >
        上海
      </span>
    </label>
    <label
      class="t-checkbox"
    >
      <input
        class="t-checkbox__former"
        name="city"
        type="checkbox"
        value="gz"
      />
      <span
        class="t-checkbox__input"
      />
      <span
        class="t-checkbox__label"
      >
        广州
      </span>
    </label>
    <label
      class="t-checkbox t-is-checked"
    >
      <input
        class="t-checkbox__former"
        name="city"
        type="checkbox"
        value="sz"
      />
      <span
        class="t-checkbox__input"
      />
      <span
        class="t-checkbox__label"
      >
        深圳
      </span>
    </label>
  </div>
</div>
`;
